config/database_test.go

```diff
@@ -415,6 +417,28 @@ func TestDatabaseStoreLoad(t *testing.T) {
 		assert.Equal(t, map[string]interface{}{"ServiceSettings": map[string]interface{}{"SiteURL": true}}, ds.GetEnvironmentOverrides())
 	})
 
+	t.Run("do not persist environment variables", func(t *testing.T) {
+		_, tearDown := setupConfigDatabase(t, minimalConfig, nil)
+		defer tearDown()
+
+		os.Setenv("MM_SERVICESETTINGS_SITEURL", "http://overridePersistEnvVariables")
+
+		ds, err := config.NewDatabaseStore(fmt.Sprintf("%s://%s", *sqlSettings.DriverName, *sqlSettings.DataSource))
+		require.NoError(t, err)
+		defer ds.Close()
+
+		_, err = ds.Set(ds.Get())
+		require.NoError(t, err)
+
+		assert.Equal(t, "http://overridePersistEnvVariables", *ds.Get().ServiceSettings.SiteURL)
+		assert.Equal(t, map[string]interface{}{"ServiceSettings": map[string]interface{}{"SiteURL": true}}, ds.GetEnvironmentOverrides())
+		// check that in memory configWithoutOverrides does not include overwritten variable
+		assert.Equal(t, "http://minimal", *config.GetConfigWithoutOverridesDS(ds).ServiceSettings.SiteURL)
```

Comment:
Can we just get rid of `GetConfigWithoutOverridesDS` altogether now? Having verified that the on-disk/in-database state and `Get()` interface state is accurate, the internal implementation details aren't really worth "testing" IMO.

Reply:
Excellent point, done!